### PR TITLE
Add GetVPCInfo method to ServiceMediator

### PR DESCRIPTION
**Pull request by zhengxiexie**
Create the common method used by other modules. When using gomonkey to test the method, encounter the problem that the mocked method doesn't take effect, fix it by adding -gcflags=all=-l option to prohibit inline optimization.

### CODE REVIEW

File: pkg/nsx/services/common/types.go [hunk 1]
```diff
@@ -69,3 +69,8 @@ var (
 	String = pointy.String // address of string
 	Int64  = pointy.Int64  // address of int64
 )
+
+type OrgProject struct {
+	Org     string
```
Comment (TaoZou1) [inline pkg/nsx/services/common/types.go:74]: OrgId， ProjectId？

Comment (zhengxiexie) [reply pkg/nsx/services/common/types.go:74]: OK

File: pkg/nsx/services/mediator/mediator.go [hunk 2]
```diff
@@ -14,3 +23,25 @@ type ServiceMediator struct {
...
+// so we return a slice of OrgProject.
+func (serviceMediator *ServiceMediator) GetOrgProjectVPC(ns string) []common.OrgProjectVPC {
+	var orgProjectVPC []common.OrgProjectVPC
+	vpcs := serviceMediator.GetVPCsByNamespace(ns) // Transparently call the VPCService.GetVPCsByNamespace method
+	for _, v := range vpcs {
+		matches := reExp.FindStringSubmatch(*v.Path)
+		if len(matches) == 4 {
+			org := matches[1]
+			project := matches[2]
+			vpc_ := matches[3]
```
Comment (lxiaopei) [inline pkg/nsx/services/mediator/mediator.go:40]: why add "_" suffix for "vpc_" variable?

File: pkg/nsx/services/mediator/mediator.go [hunk 3]
```diff
@@ -14,3 +23,25 @@ type ServiceMediator struct {
 	*securitypolicy.SecurityPolicyService
 	*vpc.VPCService
 }
+
+// GetOrgProjectVPC is a common method, extracting the org and project string from vpc path of the VPC model.
+// VPC path looks like "/orgs/default/projects/project-1/vpcs/vpc-1",
+// Since other modules only know namespace, this is the only entry point to get org and project.
+// Currently, we only support one vpc per namespace, but we may support multiple vpcs per namespace in the future,
+// so we return a slice of OrgProject.
```
Comment (lxiaopei) [inline pkg/nsx/services/mediator/mediator.go:31]: OrgProject -> OrgProjectVPC?

Comment (zhengxiexie) [reply pkg/nsx/services/mediator/mediator.go:31]: ok

File: pkg/nsx/services/mediator/mediator.go (hunk 2 above)
Comment (zhengxiexie) [reply pkg/nsx/services/mediator/mediator.go:40]: IDE reminds 
image
so add this trick.

File: pkg/nsx/services/common/types.go [hunk 4]
```diff
@@ -69,3 +69,9 @@ var (
 	String = pointy.String // address of string
 	Int64  = pointy.Int64  // address of int64
 )
+
+type OrgProjectVPC struct {
```
Comment (dantingl) [inline pkg/nsx/services/common/types.go:73]: The struct name is a little weird to me..

Comment (zhengxiexie) [reply pkg/nsx/services/common/types.go:73]: It consists of org, project, and vpc, which is a common struct used often.

File: pkg/nsx/services/mediator/mediator.go [hunk 5]
```diff
@@ -14,3 +23,25 @@ type ServiceMediator struct {
 	*securitypolicy.SecurityPolicyService
 	*vpc.VPCService
 }
+
+// GetOrgProjectVPC is a common method, extracting the org and project string from vpc path of the VPC model.
+// VPC path looks like "/orgs/default/projects/project-1/vpcs/vpc-1",
```
Comment (heypnus) [inline pkg/nsx/services/mediator/mediator.go:28]: From the comment, the input should be a string. I didn't get why you use the ns as the input, I think it's not common enough.

Comment (zhengxiexie) [reply pkg/nsx/services/mediator/mediator.go:28]: The comment explains how to extract, and the input is indeed a namespace.

**Review by dantingl — APPROVED** [review]

**Review by dantingl — APPROVED** [review]

**Pull request merged**